modelator_py/util/tlc/stdout_to_informal_trace_format.py

```diff
@@ -15,41 +15,59 @@ def trace_lines_model_checking_mode(stdout) -> typing.List[typing.List[str]]:
     """
     ret = []
     lines = stdout.split("\n")
+    header_open = False
 
     def is_header(line):
-        """Begins a trace and may also end a previous trace"""
-        HEADER = "Error: Invariant"
-        return line.startswith(HEADER)
+        """One line before the beginning of the trace."""
+        single_state_trace_header = "is violated by the initial state" in line
+        mult_state_trace_header = line == "Error: The behavior up to this point is:"
+        return single_state_trace_header or mult_state_trace_header
+
+    def is_start_of_new_trace(line):
+        """When there are multiple traces, closes the previous trace"""
+        return line.startswith("Error: Invariant")
 
     def is_footer(line):
-        return (
+        multi_state_footer = (
             ("states generated" in line)
             and ("distinct states found" in line)
             and ("states left on queue" in line)
             and (not line.startswith("Progress"))
         ) or ("Model checking completed" in line)
 
+        single_state_footer = "Finished in" in line
+
+        return single_state_footer or multi_state_footer
+
+    def format_single_state_trace(trace):
+        # we need this because for single-state trace errors ther will be no states
+        # enumerated in the output (but we need them in postprocessing)
```

Comment:
If possible, it might be better to change the postprocessing. But you're the judge.

Reply:
done